src/interfaces.ts

```diff
@@ -112,20 +112,35 @@ export interface CollectionRef {
 
 export type PropertyFilterOperator = '<' | '<=' | '>' | '>=' | ':' | '!:' | '=' | '!=';
 
+export interface PropertyFilterOperatorExtended<TokenValue> {
+  operator: PropertyFilterOperator;
+  match?: PropertyFilterOperatorMatch<TokenValue>;
+}
+
+export type PropertyFilterOperatorMatch<TokenValue> =
+  | PropertyFilterOperatorMatchByType
+  | PropertyFilterOperatorMatchCustom<TokenValue>;
+
+export type PropertyFilterOperatorMatchByType = 'date' | 'datetime';
+
+export type PropertyFilterOperatorMatchCustom<TokenValue> = (itemValue: unknown, tokenValue: TokenValue) => boolean;
```

Comment:
The itemValue type is unknown due to challenges in passing the actual type as generic.